File: framework/src/play/src/main/java/play/routing/HandlerDef.java
/*
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */

package play.routing;

import java.util.List;
import scala.collection.Seq;

import play.libs.Scala;

public abstract class HandlerDef {
  public abstract ClassLoader classLoader();
  public abstract String routerPackage();
  public abstract String controller();
  public abstract String method();
  protected abstract Seq<Class<?>> parameterTypes();
  public abstract String verb();
  public abstract String path();
  public abstract String comments();
  protected abstract Seq<String> modifiers();

  public List<Class<?>> getParameterTypes() {
    return Scala.asJava(parameterTypes());
  }

  public List<String> getModifiers() {
    return Scala.asJava(modifiers());
  }
}

File: framework/src/play/src/main/scala/play/api/routing/HandlerDef.scala
/*
 * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package play.api.routing

/**
 * Information about a `Handler`, especially useful for loading the handler
 * with reflection.
 */
case class HandlerDef(
    classLoader: ClassLoader,
    routerPackage: String,
    controller: String,
    method: String,
    parameterTypes: Seq[Class[_]],
    verb: String,
    path: String,
    comments: String = "",
    modifiers: Seq[String] = Seq.empty
) extends play.routing.HandlerDef
